### PR TITLE
feat/nui-extraction branch: Make the game playable again

### PR DESCRIPTION
**Pull request by BenjaminAmos**
### Description
This pull request depends on and should be integrated into #4017.

The code from that branch previously crashed due a type conflict in `MouseDevice` between the extracted NUI's version of `org.terasology.input` and the built-in (original) version, as NUI now uses `Vector2i` from JOML instead of TeraMath (incompatible method definitions).

I also made some fixes to prevent a crash in `NUIManagerInternal` due to it querying key bindings that don't exist yet (often when the game is started for the first time).

In addition, a serialisation bug relating to the removal of TeraMath serialisers from the codebase has been fixed, by re-adding the serialisers previously removed (as they are no longer in `nui-reflect`). This fixes the camera behaving incorrectly when a world is loaded for the first time, as well as various other subtle issues I probably overlooked.

Finally, I've converted the internal browser widgets to use JOML, to be consistent with all of the NUI built-in widgets. I also fixed the engine tests to compile again.

### How to test
In order to test gameplay, you will need to first make some changes to the core modules in the workspace (to get them to compile again). I've tried to patch as many of the omega distribution modules as I can and my code patches for those can all be found in [this](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641) gist. For the essential core modules though, some pull requests have already been issued which will fix those modules as well.

The following modules need to be fixed to test this (as a minimum):
#### Core Sample Gameplay
   - Inventory ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-inventory-patch) or https://github.com/Terasology/Inventory/pull/23)
   - Health ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-health-patch) or https://github.com/Terasology/Health/pull/44)
   - CoreWorlds ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-coreworlds-patch) or https://github.com/Terasology/CoreWorlds/pull/4)
   - CoreRendering ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-corerendering-patch) or https://github.com/Terasology/CoreRendering/pull/12)

Optionally, you might choose to test with a more complex set of modules (more fixes needed and less tested):
#### Josharias Survival
   - Josharias Survival ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-joshariassurvival-patch))
   - IRL Corp ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-irlcorp-patch))
   - Eventual Skills ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-eventualskills-patch))
   - Manual Labor ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-manuallabor-patch))
   - Manual Labor Eventual Skills ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-manuallaboreventualskills-patch))
   - CoreWorlds ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-coreworlds-patch) or https://github.com/Terasology/CoreWorlds/pull/4)
   - Worldly Tooltip ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-worldlytooltip-patch))
   - Hunger ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-hunger-patch))
   - Thirst ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-thirst-patch))
   - Durability ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-durability-patch))
   - Machines ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-machines-patch))
   - Substance Matters ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-substancematters-patch))
   - Health ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-health-patch) or https://github.com/Terasology/Health/pull/44)
   - Potential Energy Devices ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-potentialenergydevices-patch))
   - Workstation ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-workstation-patch))
   - In Game Help ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-ingamehelp-patch))
   - Basic Crafting ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-basiccrafting-patch))
   - Fluid ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-fluid-patch))
   - In Game Help API ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-ingamehelpapi-patch))
   - CoreRendering ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-corerendering-patch) or https://github.com/Terasology/CoreRendering/pull/12)
  - Inventory ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-inventory-patch) or https://github.com/Terasology/Inventory/pull/23)
   - Caves ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-caves-patch))
   - Custom Ore Gen ([patch](https://gist.github.com/BenjaminAmos/1b147c735ef22320a83bd64d49007641#file-customoregen-patch))
#### Gameplay Testing
 - Start Terasology and create a new World, selecting either the "Core Gameplay" (fewer modules) or "Josharias Survival" (more modules) template.
 - Play the game as normal. There should not be any additional crashes than were present before.

### Known Issues
 - The mouse cursor in the centre of the screen may render slightly incorrectly

### CODE REVIEW

File: engine/src/main/java/org/terasology/rendering/nui/internal/NUIManagerInternal.java [hunk 1]
```diff
@@ -145,7 +145,10 @@ public NUIManagerInternal(TerasologyCanvasRenderer renderer, Context context) {
         TabbingManager.setFocusManager(this);
 
         // NOTE: Fix for tests
-        if (bindsManager != null) {
+        if (bindsManager != null
+                && bindsManager.getBindsConfig().hasBinds(new SimpleUri("engine:tabbingUI"))
+                && bindsManager.getBindsConfig().hasBinds(new SimpleUri("engine:tabbingModifier"))
+                && bindsManager.getBindsConfig().hasBinds(new SimpleUri("engine:activate"))) {
```
Comment (keturn) [inline engine/src/main/java/org/terasology/rendering/nui/internal/NUIManagerInternal.java:151]: I'm worried about this one. If we skip over setting up this up here, when _will_ it get done?

I'd want to either fix whatever is supposed to set those bindings so it reliably runs before we get here, or move this to a callback or event handler that is called once the bindsManager is all configured.
